src/node_jsvmapi.cc

```diff
@@ -1407,34 +1365,98 @@ napi_status napi_get_value_bool(napi_env e, napi_value v, bool* result) {
   return GET_RETURN_STATUS();
 }
 
-napi_status napi_get_string_length(napi_env e, napi_value v, int* result) {
+// Gets the number of CHARACTERS in the string.
+napi_status napi_get_value_string_length(napi_env e, napi_value v, size_t* result) {
   NAPI_PREAMBLE(e);
   CHECK_ARG(result);
 
-  *result = v8impl::V8LocalValueFromJsValue(v).As<v8::String>()->Length();
+  v8::Isolate *isolate = v8impl::V8IsolateFromJsEnv(e);
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::String> str;
+
+  CHECK_TO_STRING(context, str, v);
+
+  *result = static_cast<size_t>(str->Length());
 
   return GET_RETURN_STATUS();
 }
 
-napi_status napi_get_string_utf8_length(napi_env e, napi_value v, int* result) {
+// Gets the number of BYTES in the UTF-8 encoded representation of the string.
+napi_status napi_get_value_string_utf8_length(napi_env e, napi_value v, size_t* result) {
   NAPI_PREAMBLE(e);
   CHECK_ARG(result);
 
-  *result = v8impl::V8LocalValueFromJsValue(v).As<v8::String>()->Utf8Length();
+  v8::Isolate *isolate = v8impl::V8IsolateFromJsEnv(e);
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::String> str;
+
+  CHECK_TO_STRING(context, str, v);
+
+  *result = static_cast<size_t>(str->Utf8Length());
 
   return GET_RETURN_STATUS();
 }
 
-napi_status napi_get_string_utf8(napi_env e, napi_value v, char* buf, int bufsize, int* result) {
+// Copies a JavaScript string into a UTF-8 string buffer. The result is the number
+// of bytes copied into buf, including the null terminator. If the buf size is
+// insufficient, the string will be truncated, including a null terminator.
+napi_status napi_get_value_string_utf8(
+    napi_env e, napi_value v, char* buf, size_t bufsize, size_t* result) {
+  NAPI_PREAMBLE(e);
+
+  v8::Isolate *isolate = v8impl::V8IsolateFromJsEnv(e);
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::String> str;
+
+  CHECK_TO_STRING(context, str, v);
```

Comment:
Now I'm thinking maybe these `napi_get_value_*` functions should not automatically try to *coerce* the value to a string. Rather, there should be a helper method in the C++ wrapper to do that if desired. Something like `ToString()` vs `AsString()`.